server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -851,6 +857,12 @@ private void doDeleteShardSnapshots(
                 ActionListener.wrap(() -> listener.onResponse(updatedRepoData)),
                 2
             );
+            releaseResourceLockFiles(
```

Comment:
Can we move this to per shard level operation which is done in `writeUpdatedShardMetaDataAndComputeDeletes` from line 1022 onwards ? That way it will work in parallel across shards instead of sequential behavior in current approach.

Reply:
Moved cleanup to separate method